doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+favor of a centralized policy for issue reporting. Tickets concerning, for
+example, bugs or enhancements in modules shall be opened in the main
+Zephyr repository. Issues should be appropriately labeled using Github
+labels corresponding to each module, when applicable.
+
+  .. note::
+
+     Bug reports in modules, mirroring upstream project bug reports, are
+     allowed but shall affect the
+     :ref:`Release Quality Criteria<release_quality_criteria>`.
+
+
+.. _modules_licensing:
+
+Licensing requirements and policies
+***********************************
+
+All source files in a module's code base require to include a license header,
+unless there exist a **main license file** that covers source files that do
+not possess a license header.
+
+Main license files shall be added in the module code base by Zephyr
+developers, only if they exist as part of the external project,
+and they contain a permissive OSI-compliant license. Main license files
+should preferably contain a full license text instead of including an
+SPDX license identifier. If multiple main license files are present it
+shall be made clear which license applies to each source file in a module
+code base.
+
+Individual license headers in module source files supersede the main license.
+
+Any new content to be added in a module repositotry will require to have
+license coverage.
+
+  .. note::
+
+     Zephyr recommends conveying module licensing via individual license
+     headers and main license files. This not a hard requirement; should
+     an external project have its own practice of conveying how licensing
+     applies in the module code base, this practice may be accepted by and
+     be referred to in the Zephyr module, as long as licensing requirement,
+     for example OSI compliance, are satisfied.
+
+License policies
+================
+
+When creating a module repository a developer shall
+
+* import the main license files, if they exist in the external project, and
+* document (for example in the module README or .yml file) the default license
+  that covers the module code base.
+
+License checks
+--------------
+
+License checks (via CI tools) shall be enabled on every pull request that
+adds new content in module repositories.
+
+
+Documentation requirements
+**************************
+
+All Zephyr module repositories shall include an .rst file documenting
+
+* the scope and the purpose of the module
+* how the module integrates with Zephyr
+* the code owner of the module repository
+* synchronization information with the external project (commit, SHA, version etc.)
+* licensing information as described in :ref:`modules_licensing`.
+
+The file shall be required for the inclusion of the module and the contained
+information should be kept up to date.
+
+
+Testing requirements
+********************
+
+All Zephyr modules should provide some level of **integration** testing,
+ensuring that the integration with Zephyr works correctly.
+Integration tests
```

Comment:
```suggestion
Integration tests:
```

Several bullet lists in this PR do not have a colon (:) in the text introducing them. Can you please check and add the missing colons?

Reply:
will do, thanks.